File: leetcode/0300-0399/338. Counting Bits/338.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0] * (n + 1)
        for i in range(1, n + 1):
            j = i
            count = 0
            while j:
                j = j & (j - 1)
                count += 1
            res[i] = count
        return res

File: leetcode/0300-0399/338. Counting Bits/338.cs
public class Solution
{
    public int[] CountBits(int n)
    {
        int[] res = new int[n + 1];
        for (int i = 1; i <= n; i++)
        {
            int j = i, count = 0;

            while (j != 0)
            {
                j = j & (j - 1);
                count++;
            }
            res[i] = count;
        }
        return res;
    }
}